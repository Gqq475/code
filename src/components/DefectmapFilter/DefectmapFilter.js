import React from 'react'
import classes from './DefectmapFilter.scss'
import DatePicker2 from 'components/DatePicker2'
import AvailableFilters from 'components/AvailableFilters'
import { Button } from '@blueprintjs/core'
import { dateHOC } from '../../Utils'
import { assign } from 'lodash'

const formatDate = (date) => date.format('YYYY-MM-DD HH:mm:ss')

type Props = {
  getLots: Function,
  getGlasses: Function,
  dateStart: String,
  dateEnd: String
}
export class DefectmapFilter extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.state = {
      product: 'SV509-CF',
      operation: '2100',
      lineID: 'PIPR01',
      subEntity: 'PIN01',
      pageNo: 1,
      // Must set 1 to get total count
      pageSize: 1,
      sortBy: 'id',
      order: 'asc'
    }
  }

  handleClick () {
    this.props.getLots(assign({}, this.state, {
      start_time: formatDate(this.props.dateStart),
      end_time: formatDate(this.props.dateEnd)
    }))
    this.props.getGlasses(assign({}, this.state, {
      start_time: formatDate(this.props.dateStart),
      end_time: formatDate(this.props.dateEnd)
    }))
  }

  componentDidMount () {
    this.handleClick()
  }

  render () {
    return (
      <div className={classes['DefectmapFilter-container']}>
        <DatePicker2 {...this.props} />
        <AvailableFilters {...this.state} that={this} />
        <Button className={classes['btn-submit']} onClick={this.handleClick}>
          查询
        </Button>
      </div>
    )
  }
}

export default dateHOC(props => <DefectmapFilter {...props} />)
