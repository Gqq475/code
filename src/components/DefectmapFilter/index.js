import DefectmapFilter from './DefectmapFilter'
export default DefectmapFilter
