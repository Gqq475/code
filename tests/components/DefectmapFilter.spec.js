import React from 'react'
import { DefectmapFilter } from 'components/DefectmapFilter/DefectmapFilter'

describe('(Component) DefactJudgeFilter', () => {
  it('render normally', () => {
    let props = {
      getLots: jest.fn(),
      getGlasses: jest.fn()
    }
    const wrapper = shallow(
      <DefectmapFilter {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
