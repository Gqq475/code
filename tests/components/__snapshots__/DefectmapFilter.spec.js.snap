// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DefactJudgeFilter render normally 1`] = `
<div
  className="DefectmapFilter-container"
>
  <DatePicker2
    getGlasses={[Function]}
    getLots={[Function]}
  />
  <AvailableFilters
    lineID="PIPR01"
    operation="2100"
    order="asc"
    pageNo={1}
    pageSize={1}
    product="SV509-CF"
    sortBy="id"
    subEntity="PIN01"
    that={
      DefectmapFilter {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <DefectmapFilter
            getGlasses={[Function]}
            getLots={[Function]}
      />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="DefectmapFilter-container"
      >
              <DatePicker2
                      getGlasses={[Function]}
                      getLots={[Function]}
              />
              <AvailableFilters
                      lineID="PIPR01"
                      operation="2100"
                      order="asc"
                      pageNo={1}
                      pageSize={1}
                      product="SV509-CF"
                      sortBy="id"
                      subEntity="PIN01"
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div
              className="DefectmapFilter-container"
      >
              <DatePicker2
                      getGlasses={[Function]}
                      getLots={[Function]}
              />
              <AvailableFilters
                      lineID="PIPR01"
                      operation="2100"
                      order="asc"
                      pageNo={1}
                      pageSize={1}
                      product="SV509-CF"
                      sortBy="id"
                      subEntity="PIN01"
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "handleClick": [Function],
        "props": Object {
          "getGlasses": [Function],
          "getLots": [Function],
        },
        "refs": Object {},
        "state": Object {
          "lineID": "PIPR01",
          "operation": "2100",
          "order": "asc",
          "pageNo": 1,
          "pageSize": 1,
          "product": "SV509-CF",
          "sortBy": "id",
          "subEntity": "PIN01",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
  />
  <Blueprint.Button
    className="btn-submit"
    onClick={[Function]}
  >
    查询
  </Blueprint.Button>
</div>
`;
